crates/sui-types/src/message_envelope.rs

```diff
@@ -224,7 +226,6 @@ where
     // TODO: Eventually we should remove all calls to verify_signature
     // and make sure they all call verify to avoid repeated verifications.
     pub fn verify_signature(&self, committee: &Committee) -> SuiResult {
-        self.data.verify(Some(self.auth_sig().epoch))?;
```

Comment:
why was this removed?

Reply:
sorry that meant to be highlighted instead of deleted. this method is invoked at many places that i dont have access to state.jwk_as_bytes, most of them are in the client side (safe client, authority aggregator). do we need a zk login tx pass this verification or its ok to skip?

Reply:
I don't think we should skip verification entirely on the client. It would probably be best if clients fetched the JWK themselves, which should be quite simple to do. I think its okay if they aren't sync with the validators though, they can just fetch directly from google.

The reasoning here is that zklogin verification is quite a bit more expensive than previous authenticators, and is potentially a DoS attack surface. So we may need to rate limit or ban clients that send invalid zklogins. But the "client" from the validators point of view is usually a fullnode. So this in turn means that fullnodes would need to ensure sigs are valid before sending on to the validators, or end users could get the fullnodes banned.